airbyte-ci/connectors/metadata_service/lib/tests/test_validators/test_metadata_validators.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+
+import pytest
+import requests
+import semver
+import yaml
+from metadata_service.models.generated.ConnectorMetadataDefinitionV0 import ConnectorMetadataDefinitionV0
+from metadata_service.validators import metadata_validator
```

Comment:
I'd love to have some unit tests here that don't do a network round trip, and that also handle the case where there's no `latest`.

Reply:
> Handle the case where there's not `latest`

This case is handled, if no latest version is found we pass the validation as it means the connector has never been published yet:
https://github.com/airbytehq/airbyte/blob/augustin/07-30-metadata-service_validate_that_the_dockerImageTag_is_not_decremented/airbyte-ci/connectors/metadata_service/lib/metadata_service/validators/metadata_validator.py#L189

Reply:
I added a `test_validation_pass_on_docker_image_no_latest` test. 
I'd prefer to keep the network round trip as if the test fail it might characterize a regression in DockerHub API we'd want to address.

Reply:
I'm fine having tests with a round trip but think that a separate test where the values for the versions are deterministic is appropriate here.

Reply:
Great, thanks!